content/en/docs/concepts/services-networking/topology-aware-hints.md

```diff
@@ -0,0 +1,127 @@
+to endpoints with hints for the zone it is in.
+
+## Safeguards
+
+Both the EndpointSlice controller and Kube-Proxy implementations have a number
+of safeguards built in that will result in falling back to the original
+approach. This results in traffic being routed to all endpoints, instead of just
+a topology aware subset. Here's what could trigger that:
+
+1. **Insufficient number of endpoints:** If there are less endpoints than zones
+   in a cluster, the controller will not assign any hints. 
+
+2. **Impossible to achieve balanced allocation:** In some cases, it will be
+   impossible to achieve a balanced allocation of endpoints among zones. For
+   example, if zone-a is twice as large as zone-b, but there are only 2
+   endpoints, an endpoint allocated to zone-a may receive twice as much traffic
+   as zone-b. The controller wil not assign hints if it can't get this "expected
+   overload" value below 20% for each zone. Importantly this is not based on
+   real-time feedback. It is still possible for individual endpoints to become
+   overloaded.
+
+3. **One or more Nodes has insufficient information:** If a node does not have a
+   `topology.kubernetes.io/zone` label or is not reporting a value for
+   allocatable CPU, this functionality will be disabled for all Services in a
+   cluster.
+
+4. **One or more endpoints does not have a zone hint:** When this happens,
+   kube-proxy assumes that a transition from or to Topology Aware Hints is
+   underway. Filtering endpoints for a Service in this state would be dangerous
+   so Kube-Proxy falls back to using all endpoints.
+
+5. **A zone is not represented in hints:** If kube-proxy is unable to find at
+   least one endpoint with a hint targeting the zone it is running in, it will
+   fall back to using endpoints from all zones. This is most likely to happen as
+   a new zone is being added to a cluster.
+
+## Constraints
+
+* Topology Aware Hints are not compatible with `internalTrafficPolicy=Local`,
+  and therefore a Service cannot use both of these features. It is possible to
+  use both features in the same cluster on different Services, just not on the
+  same Service.
+
+* This approach will not work well for Services that have a large proportion of
+  traffic originating from a subset of zones. Instead this assumes that incoming
+  traffic will be roughly proportional to the capacity of the Nodes in each
+  zone.
+
+* The EndpointSlice controller ignores unready nodes as it calculates the
+  proportions of each zone. This could have unintended consequences if a large
+  portion of nodes are unready.
+
+* The EndpointSlice controller does not take into account tolerations when
+  deploying calculating the proportions of each zone. If the Pods backing a
+  Service are limited to a subset of Nodes in the cluster, this will not be
+  taken into account.
+
+* This may not work well with autoscaling. For example, if a lot of traffic is
```

Comment:
Is this correct:
```suggestion
* Topology Aware hints may not work well if you use use horizontal autoscaling at either the pod or cluster level. For example, if a lot of traffic is
```
?

Reply:
I think this is primarily concerned with horizontal autoscaling at the pod level, but I think this could also interact poorly with vertical pod autoscaling, but I don't have as much experience there.

Reply:
Feel free to pick any wording - emphasis on HPA makes sense given what you've said here.

Reply:
Good call, updated to add a reference to HPA here.